### PR TITLE
fix: implement 'login' event for WebContents

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
The 'login' event was disabled during the Great Network Servicification (see https://github.com/electron/electron/issues/19602). This reimplements it on top of the new stuff.

Fixes https://github.com/electron/electron/issues/20752

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed the 'login' event not being emitted when HTTP Basic auth was requested from a WebContents.

### CODE REVIEW

**Comment by nornagon** [comment]
TODO: this event this commit implements doesn't include request details (url, referrer, method, etc.), because they aren't easily accessible at the time the event is emitted. We should either figure out a way to get that info or update the docs to reflect that the info is no longer present.

**Comment by nornagon** [comment]
Also TODO: events from the 'net' module...

File: spec-main/api-web-contents-spec.ts [hunk 1]
```diff
@@ -1518,4 +1518,37 @@ describe('webContents module', () => {
...
+        serverUrl = 'http://127.0.0.1:' + (server.address() as AddressInfo).port
+        done()
+      })
+    })
+
+    after(() => {
+      server.close()
+    })
+
+    it('receives login event on webcontents', (done) => {
```
Comment (MarshallOfSound) [inline spec-main/api-web-contents-spec.ts]: Can we make this test actually verify the `cb` worked.  Like callback with valid creds and test the page loaded and callback with invalid creds and assert it didn't

Comment (nornagon) [reply spec-main/api-web-contents-spec.ts]: done

**Comment by nornagon** [comment]
Since the issue with the `net` module is blocked waiting on a response from upstream, i think it would be a good idea to merge this and not wait.

**Comment by gavinkalika** [comment]
any movement on getting this fixed merged

**Pull request merged**